cdap-common/src/main/java/io/cdap/cdap/common/internal/remote/TaskWorkerHttpHandlerInternal.java

```diff
@@ -115,17 +124,20 @@ public TaskWorkerHttpHandlerInternal(CConfiguration cConf,
     this.metricsCollectionService = metricsCollectionService;
     this.metadataServiceEndpoint = cConf.get(
         Constants.TaskWorker.METADATA_SERVICE_END_POINT);
+    this.runningTasks = new ConcurrentHashMap<>();
     boolean enableUserCodeIsolationEnabled = cConf.getBoolean(
         TaskWorker.USER_CODE_ISOLATION_ENABLED);
     if (enableUserCodeIsolationEnabled) {
       // Run only one request at a time in user code isolation mode.
       this.requestLimit = 1;
       // Restart the service to clean up and re-claim resources after user code
       // execution.
-      this.taskCompletionConsumer = (succeeded, taskDetails) -> {
+      this.taskCompletionConsumer = (succeeded, runningTaskDetails) -> {
+        TaskDetails taskDetails = runningTaskDetails.getTaskDetails();
         taskDetails.emitMetrics(succeeded);
         runningRequestCount.decrementAndGet();
         requestProcessedCount.incrementAndGet();
+        runningTasks.remove(runningTaskDetails);
```

Comment:
Do we still need requestProcessedCount now that we have runningTaks map?

Reply:
Reverted most of the changes to use a periodic restart approach. This map is not present anymore.